sources/packages/api/src/route-controllers/assessment/assessment.controller.service.ts

```diff
@@ -273,4 +287,124 @@ export class AssessmentControllerService {
   }
+
+  /**
+   * Get all requests related to an assessments for a student
+   * application, i.e, this will fetch all pending and denied
+   * student appeals for an application or possible application
+   * exceptions that will prevent the assessment to proceed till
+   * they are approved, for instance, when a document is uploaded
+   * and need to be reviewed.
+   * @param applicationId application number.
+   * @param studentId applicant student.
+   * @param  includeOnlyAppeals then only student appeals are returned and
+   * other assessment types must not be available when this property is true.
+   * @returns assessment requests or exceptions for a student application.
+   */
+
+  async getRequestedAssessmentSummary(
+    applicationId: number,
+    studentId?: number,
+    includeOnlyAppeals = false,
+  ): Promise<RequestAssessmentSummaryAPIOutDTO[]> {
+    const requestAssessmentSummary: RequestAssessmentSummaryAPIOutDTO[] = [];
+    // Requested offering changes and application exceptions must not be included when includeOnlyAppeals is true.
+    if (!includeOnlyAppeals) {
+      const offeringChange =
+        await this.educationProgramOfferingService.getOfferingRequestsByApplicationId(
+          applicationId,
+        );
+      if (offeringChange) {
+        requestAssessmentSummary.push({
+          id: offeringChange.id,
+          submittedDate: offeringChange.submittedDate,
+          status: offeringChange.offeringStatus,
+          requestType: RequestAssessmentTypeAPIOutDTO.OfferingRequest,
+          programId: offeringChange.educationProgram.id,
+        });
+      }
+
+      const applicationExceptions =
+        await this.applicationExceptionService.getExceptionsByApplicationId(
+          applicationId,
+          ApplicationExceptionStatus.Pending,
+          ApplicationExceptionStatus.Declined,
+        );
+
+      if (applicationExceptions.length > 0) {
+        const applicationExceptionArray: RequestAssessmentSummaryAPIOutDTO[] =
+          applicationExceptions.map((applicationException) => ({
+            id: applicationException.id,
+            submittedDate: applicationException.createdAt,
+            status: applicationException.exceptionStatus,
+            requestType: RequestAssessmentTypeAPIOutDTO.StudentException,
+          }));
+        return requestAssessmentSummary.concat(applicationExceptionArray);
+      }
+    }
+
+    const studentAppeal =
+      await this.studentAppealService.getPendingAndDeniedAppeals(
+        applicationId,
+        studentId,
+      );
+    const studentAppealArray: RequestAssessmentSummaryAPIOutDTO[] =
+      studentAppeal.map((appeals) => ({
+        id: appeals.id,
+        submittedDate: appeals.submittedDate,
+        status: appeals.status,
+        requestType: RequestAssessmentTypeAPIOutDTO.StudentAppeal,
+      }));
+    return requestAssessmentSummary.concat(studentAppealArray);
+  }
+
+  /**
+   * Method to get history of assessments for an application,
```

Comment:
I would prefer to remove from the comments the sentence part that states of the code is like: "Method", "Class", "Endpoint", and "Property". Without it is as meaningful, as shown below.
```
Get the history of assessments for an application
```